shared/js/sidebar.jsx

```diff
@@ -0,0 +1,179 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var loop = loop || {};
+
+loop.sidebar = (function(mozL10n) {
+  "use strict";
+
+  var ROOM_STATES = loop.store.ROOM_STATES;
+  var sharedActions = loop.shared.actions;
+  var sharedUtils = loop.shared.utils;
+  var sharedMixins = loop.shared.mixins;
+
+  var SidebarControllerView = React.createClass({
+    mixins: [sharedMixins.UrlHashChangeMixin,
+             sharedMixins.DocumentLocationMixin,
+             loop.store.StoreMixin("activeRoomStore"),
+             Backbone.Events],
+
+    propTypes: {
+      activeRoomStore: React.PropTypes.instanceOf(loop.store.ActiveRoomStore).isRequired,
+      dispatcher: React.PropTypes.instanceOf(loop.Dispatcher).isRequired
+    },
+
+    getInitialState: function() {
+      return this.getStoreState();
+    },
+
+    componentWillMount: function() {
+      this.props.activeRoomStore.on("change", function() {
```

Comment:
This component is doing a mixture of using the StoreMixin and using the old style listens. I think you should be able to drop the `getInitialState`, `componentWillMount` and `componentWillUnmount` functions here.